tests/integration/suite/test_auth_proxy.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+import rancher
+import requests
+from os import path
+
+certs_exist = pytest.mark.skipif(
+    not path.exists("/etc/rancher/ssl/failclient.pem"),
+    reason='Test certs dont exist for proxy'
+)
+
+
+@certs_exist
+def test_user_proxy(admin_proxy):
+    headers = {
+        "X-Remote-User": admin_proxy.user.id,
+        "X-Remote-Group": "abc"
+    }
+    certs = ('/etc/rancher/ssl/client.pem')
+    client = rancher.Client(url=admin_proxy.BASE_URL, verify=False,
+                            headers=headers, cert=certs)
+    client.list_user(username='admin').data[0]
+    assert True
+
+
+@certs_exist
+def test_user_proxy_invalid_cert(admin_proxy):
+    headers = {
+        "X-Remote-User": admin_proxy.user.id,
+        "X-Remote-Group": "abc"
+    }
+    certs = ('/etc/rancher/ssl/failclient.pem')
+    with pytest.raises(requests.exceptions.RequestException) as e:
+        rancher.Client(url=admin_proxy.BASE_URL, verify=False,
+                       headers=headers, cert=certs)
+    assert e is not None
```

Comment:
Can this check be more specific? What error are we expecting back

Reply:
Its the requests package throwing an error, I made it vague because I was assuming it could blow up for several reasons in those underlying packages (requests/urllib vs rancher), but I'll try to narrow it down.